Form/Extension/DisableCSRFExtension.php

```diff
@@ -22,22 +22,18 @@
  */
 class DisableCSRFExtension extends AbstractTypeExtension
 {
-    private $securityContext;
+    private $authorizationChecker;
     private $role;
 
-    public function __construct(SecurityContextInterface $securityContext, $role)
+    public function __construct(AuthorizationChecker $authorizationChecker, $role)
     {
-        $this->securityContext = $securityContext;
+        $this->authorizationChecker = $authorizationChecker;
         $this->role = $role;
     }
 
-    public function setDefaultOptions(OptionsResolverInterface $resolver)
+    public function configureOptions(OptionsResolver $resolver)
     {
-        if (!$this->securityContext->getToken()) {
-            return;
-        }
-
-        if (!$this->securityContext->isGranted($this->role)) {
+        if (!$this->authorizationChecker->isGranted($this->role)) {
```

Comment:
this will throw an exception in case there is no token (i.e. outside of a firewall), which is why there was the check on the token you removed